pkg/fanotify/conn/conn.go

```diff
@@ -8,16 +8,29 @@ package conn
 
 import (
 	"bufio"
-	"io"
+	"encoding/json"
 )
 
 type Client struct {
-	Scanner *bufio.Scanner
+	Reader *bufio.Reader
 }
 
-func (c *Client) GetPath() (string, error) {
-	if !c.Scanner.Scan() { // NOTE: no timeout
-		return "", io.EOF
+type EventInfo struct {
+	Path      string `json:"path"`
+	Size      uint32 `json:"size"`
+	Timestamp int64  `json:"timestamp"`
+}
+
+func (c *Client) GetEventInfo() ([]EventInfo, error) {
+	data, err := c.Reader.ReadString('\n')
```

Comment:
Why must wait for `\n` ?

Reply:
It's better to wait for EOF here

Reply:
Fixed. 

The `bufio.Reader.ReadBytes` is not limited by the buffer size. But it needs to take a `delim` argument, so we pass the '\n'.
Before reaching '\n', the reader has successfully read the event information from the optimizer server because of `io.EOF`.